### DISCUSSION TITLE
BSON::ObjectId.from_time(Time.new(2024, 10, 1), unique: true).generation_time != Time.new(2024, 10, 1)

### DISCUSSION THREAD

**Discussion by priceline-rosenfeld** [mongodb/bson-ruby#337] [General]
The provided time doesn't seem to be respected by `ObjectId.from_time` when `unique: true` is provided.

**Comment by jamis** [comment]
Thanks for pointing this out. I've created a ticket (here: https://jira.mongodb.org/browse/RUBY-3553) to track this issue.

**Reply by priceline-rosenfeld** [reply]
Thanks!

**Comment by bnorton** [comment]
found this issue too when upgrading from 4.15.0 to 5.0.2

The issue was [deleting the handling of the time argument](https://github.com/mongodb/bson-ruby/commit/9484d2a0caa1414c4ec1e8cfb6cddf42e6e71cd4#diff-be0441e7cc9b254aae20ccb8e6117c261de5f26acc44914540ad48b2f4a53139L40-L45) to the `rb_bson_object_id_generator_next` function
